xarray/tests/test_dask.py

```diff
@@ -182,6 +183,22 @@ def test_bivariate_ufunc(self):
         self.assertLazyAndAllClose(np.maximum(u, 0), xu.maximum(v, 0))
         self.assertLazyAndAllClose(np.maximum(u, 0), xu.maximum(0, v))
 
+    def test_compute_args(self):
+        a = Variable('x', [1, 2]).chunk()
+        expected = Variable('x', [1, 4])
+        b = a * a
+        # compute
+        b1 = b.compute(get=dask.multiprocessing.get)
```

Comment:
The clean way to test this is probably with [mock](https://docs.python.org/3/library/unittest.mock.html), e.g.,
```
In [63]: import numpy as np

In [64]: import dask.array as da

In [65]: from unittest import mock

In [66]: x = da.from_array(np.arange(3), chunks=(2,))

In [67]: with mock.patch.object(da.Array, 'compute', return_value=np.arange(3)) as mock_compute:
    ...:     x.compute(foo='bar')
    ...:

In [68]: mock_compute.assert_called_with(foo='bar')

In [69]: mock_compute.assert_called_with(bar='foo')
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-69-22937cf26ca1> in <module>()
----> 1 mock_compute.assert_called_with(bar='foo')

~/conda/envs/xarray-dev/lib/python3.5/unittest/mock.py in assert_called_with(_mock_self, *args, **kwargs)
    792         if expected != actual:
    793             cause = expected if isinstance(expected, Exception) else None
--> 794             raise AssertionError(_error_message()) from cause
    795
    796

AssertionError: Expected call: compute(bar='foo')
Actual call: compute(foo='bar')
```

`unittest.mock` is part of Python 3's standard library, but there's also a widely used Python 2 backport on pypi. I think it would be perfectly fine to add it as a dependency for our test suite.

Reply:
Thanks @shoyer, I'll add mock as part of this PR.